PyMca5/PyMcaGui/io/QTiledCatalogSelectorDialog.py

```diff
@@ -173,6 +177,34 @@ def reset_url_entry(self) -> None:
         else:
             self.url_entry.setText(self.model.url)
 
+    def _rebuild_current_path_layout(self):
+        bc_widget = ClickableIndexedQLabel("root", index=0)
+        bc_widget.clicked.connect(self._on_breadcrumb_clicked)
+        breadcrumbs = [bc_widget]
+
+        for i, node_id in enumerate(self.model.node_path_parts, start=1):
+            if len(node_id) > self.NODE_ID_MAXLEN:
+                short_node_id = node_id[: self.NODE_ID_MAXLEN - 3] + "..."
+                bc_widget = ClickableIndexedQLabel(short_node_id, index=i)
+            else:
+                bc_widget = ClickableIndexedQLabel(node_id, index=i)
+
+            bc_widget.clicked.connect(self._on_breadcrumb_clicked)
+            breadcrumbs.append(bc_widget)
+        
+        # remove all widgets from current_path_layout
+        self.remove_layout_widgets()
+
+        for breadcrumb in breadcrumbs:
+            self.current_path_layout.addWidget(breadcrumb)
+            self.current_path_layout.addWidget(QLabel(" / "))
+
+    def remove_layout_widgets(self):
+        for index in reversed(range(self.current_path_layout.count())):
```

Comment:
Using a more specific name here to differentiate from the many layout widgets in the dialog.

```suggestion
    def remove_current_path_layout_widgets(self):
        """Remove unneeded path widgets and free the memory."""
```